server/src/api/modules/users/explore.ts

```diff
@@ -2,44 +2,69 @@ import { FastifyInstance } from 'fastify';
 import { prisma } from '../../../prisma';
 
 type GetUserExploreResponse = {
-  id: string;
-  stacksAddress: string;
-}[];
+  nextPage: number | null;
+  data: { id: string; stacksAddress: string }[];
+};
 const getUserExploreResponseSchema = {
-  type: 'array',
-  items: {
-    type: 'object',
-    required: ['id', 'stacksAddress'],
-    properties: {
-      id: { type: 'string' },
-      stacksAddress: { type: 'string' },
+  type: 'object',
+  required: ['data'],
+  properties: {
+    nextPage: { type: 'number', nullable: true },
+    data: {
+      type: 'array',
+      items: {
+        type: 'object',
+        required: ['id', 'stacksAddress'],
+        properties: {
+          id: { type: 'string' },
+          stacksAddress: { type: 'string' },
+        },
+      },
     },
   },
 };
 
 export async function createGetUserExploreEndpoint(fastify: FastifyInstance) {
   return fastify.get<{
     Reply: GetUserExploreResponse;
+    Querystring: { page?: number };
   }>(
     '/api/users/explore',
     {
       onRequest: [fastify.authenticate],
       schema: {
         description: 'Return a list of users using Sigle.',
         tags: ['user'],
+        querystring: {
+          page: { type: 'number' },
+        },
         response: {
           200: getUserExploreResponseSchema,
         },
       },
     },
     async (req, res) => {
+      let { page } = req.query;
+      if (!page || page === 0) {
+        page = 1;
+      }
+      const pageSize = 50;
+      const where = { stacksAddress: { not: req.address } };
+
       const users = await prisma.user.findMany({
         // Remove the current logged in user from the list
-        where: { stacksAddress: { not: req.address } },
+        where,
         orderBy: { followers: { _count: 'desc' } },
-        take: 150,
+        skip: pageSize * (page - 1),
+        take: pageSize,
+        include: {
+          _count: true,
+        },
       });
-      return res.send(users);
+      const usersCount = await prisma.user.count({ where });
+      const nextPage = usersCount > pageSize * page ? page + 1 : null;
```

Comment:
here we tell the client if there is more data to load or if the end of the list is reached